aiinfra-cluster/modules/vm-instance-group/main.tf

```diff
@@ -25,6 +25,12 @@ locals {
   enable_gvnic  = var.bandwidth_tier != "not_enabled"
   enable_tier_1 = var.bandwidth_tier == "tier_1_enabled"
 
+  is_debian_11_image = (var.instance_image.name != null && length(regexall("debian-11", var.instance_image.name)) > 0) || (var.instance_image.family != null && length(regexall("debian-11", var.instance_image.family)) > 0)
+  
+  // Terraform does not provide a way to validate multiple variables in variable validation block.
+  // Using this type of validation as per https://github.com/hashicorp/terraform/issues/25609#issuecomment-1057614400
+  validate_basic_node_pool = (!local.is_debian_11_image && (local.enable_gvnic || local.enable_tier_1)) ? tobool("GVNIC is only supported in debian 11 image.") : true
```

Comment:
this is clever!

should the error message also include something about `enable_tier_1` or is that implied in `GVNIC`?

Reply:
enable_tier_1 is GVNIC + tier_1. So GVNIC is implied with that.